internal/sigstore/sign.go

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+package sigstore
+
+import (
+	"bytes"
+	"encoding/base64"
+
+	"github.com/sigstore/cosign/v2/pkg/cosign"
+)
+
+// SignContent signs the content with the cosign encrypted private key and corresponding cosign password.
+func SignContent(password, encryptedPrivateKey, content []byte) ([]byte, error) {
+	sv, err := cosign.LoadPrivateKey(encryptedPrivateKey, password)
+	if err != nil {
+		return nil, err
+	}
+	sig, err := sv.SignMessage(bytes.NewReader(content))
+    if err != nil {
+        return nil, fmt.Errorf("signing message: %w", err)
+    }
+	return []byte(base64.StdEncoding.EncodeToString(sig)), nil
+}
```

Comment:
From a (high level) look at the cosign go package, it looks to me like this function signs the payload but doesn't upload the result to the transparency log.

I see two possible changes here:

- decide signatures created by this functions don't end up in the transparency log
  - i would make this very clear by naming the function accordingly and even adding a comment
  - I would use go stdlib code to create the signature instead. We get a large amount of new dependencies through the cosign package
-  add the transparency log upload functionality (either as a separate function or in here)

Reply:
Yes, it's not uploaded to the log. Is it okay or should I add it? But if we upload to the log, we likely need cosign as a dependency @daniel-weisse @derpsteb 
For now, I copied the function to remove the dep on cosign.

Reply:
should I open a ticket in Azure and leave as is in this PR for now? Or how to proceed?

Reply:
A ticket sounds like a good idea so we don't block this PR.

Reply:
please also add the ticket as a TODO to the function.